preview/__test__/clean.test.js

```diff
@@ -0,0 +1,156 @@
+'use strict';
+
+const Cleaner = require('../clean.js');
+const fs = require('fs').promises;
+const nock = require('nock');
+const os = require('os');
+const path = require('path');
+const rimraf = require('rimraf');
+
+const tmp = `${os.tmpdir()}/preview_cleaner`;
+
+async function prepare_cleaner(extra_branches = []) {
+  await fs.mkdir(tmp, {recursive: true});
+
+  // Create an empty repo
+  const repo = await fs.mkdtemp(`${tmp}/input`);
+  const opts = {
+    cwd: repo,
+    env: {
+      'GIT_AUTHOR_NAME': 'Test',
+      'GIT_AUTHOR_EMAIL': 'test@example.com',
+      'GIT_COMMITTER_NAME': 'Test',
+      'GIT_COMMITTER_EMAIL': 'test@example.com',
+    },
+  };
+  await Cleaner.exec_git(['init'], opts);
+  await Cleaner.exec_git(['commit', '--allow-empty', '-m', 'init'], opts);
+  for (let branch of extra_branches) {
+    await Cleaner.exec_git(['branch', branch], opts);
+  }
+
+  // Create the local cache which is required by the cleaner scripit
+  const cache_dir = await fs.mkdtemp(`${tmp}/cache`);
+  opts.cwd = cache_dir;
+  await Cleaner.exec_git(['clone', '--mirror', repo], opts);
+
+  // Setup the cleaner
+  const tmp_dir = await fs.mkdtemp(`${tmp}/tmp`);
+  return new Cleaner(null, repo, cache_dir, tmp_dir);
+}
+function mock_github(request_body) {
+  return nock('https://api.github.com').post('/graphql', request_body);
+}
+function github_result(is_closed) {
+  return JSON.stringify({
+    data: {
+      repository: {
+        pullRequest: {
+          closed: is_closed
+        }
+      }
+    }
+  });
+}
+
+afterAll(() => {
+  return new Promise((resolve, reject) => {
```

Comment:
Any reason to not use async/await in the places you're using `new Promise`? If it's just to catch an error the equivalent would be wrapping the await in a `try {} catch {}`

Reply:
I talked to @haywoood about this. There isn't actually a promise oriented API here. I'm going to see if I can switch to `rmfr` which has one.